### PR TITLE
Build Kibana Deployment pod template on top of user-provided podTemplate

### PR DESCRIPTION
**Pull request by sebgl**
This commit changes the way we build the Kibana pod template to inherit
from a user-provided template.

It allows users to set their own labels, annotations, environment
variables, volumes, etc.

Note that it does not attempt to deal with conflicting items
(environment variables, volumes, etc.). To be done in a follow-up PR.

It also does not deal with improving (fixing) the Kibana spec comparison
mechanism: to be handled with
https://github.com/elastic/cloud-on-k8s/issues/772.

It also includes some minor refactoring:
* move labels into their own package to avoid circular imports
* get rid of DeploymentParams that just propagates more and more fields
from the Kibana spec: just use the spec directly
* add a helper function to retrieve the Kibana container from the spec,
instead of operating on all containers of that spec

### CODE REVIEW

File: operators/pkg/controller/kibana/version/version7/podspecs_test.go [hunk 1]
```diff
@@ -26,26 +25,29 @@ func envWithName(t *testing.T, name string, container corev1.Container) corev1.E
 func TestNewPodSpec(t *testing.T) {
```
Comment (thbkrkr) [inline operators/pkg/controller/kibana/version/version7/podspecs_test.go]: TestNewPodSpec could be renamed to TestNewPodTemplateSpec.

File: operators/pkg/controller/kibana/version/version6/podspecs_test.go [hunk 2]
```diff
@@ -26,26 +25,29 @@ func envWithName(t *testing.T, name string, container corev1.Container) corev1.E
 func TestNewPodSpec(t *testing.T) {
```
Comment (thbkrkr) [inline operators/pkg/controller/kibana/version/version6/podspecs_test.go]: TestNewPodSpec could be renamed to TestNewPodTemplateSpec.

File: operators/pkg/controller/kibana/deployment_control.go [hunk 3]
```diff
@@ -64,16 +59,18 @@ func ReconcileDeployment(c k8s.Client, s *runtime.Scheme, expected appsv1.Deploy
...
 				!reflect.DeepEqual(expected.Spec.Template.ObjectMeta, reconciled.Spec.Template.ObjectMeta) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Name, reconciled.Spec.Template.Spec.Containers[0].Name) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Env, reconciled.Spec.Template.Spec.Containers[0].Env) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Image, reconciled.Spec.Template.Spec.Containers[0].Image) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Resources, reconciled.Spec.Template.Spec.Containers[0].Resources)
+				!reflect.DeepEqual(expectedContainer.Name, actualContainer.Name) ||
+				!reflect.DeepEqual(expectedContainer.Env, actualContainer.Env) ||
+				!reflect.DeepEqual(expectedContainer.Image, actualContainer.Image) ||
+				!reflect.DeepEqual(expectedContainer.Resources, actualContainer.Resources)
 			// TODO: do something better than reflect.DeepEqual above?
```
Comment (thbkrkr) [inline operators/pkg/controller/kibana/deployment_control.go]: This is a bit out-of-scope of this PR. I think we miss comparing several fields that the user can fill. We definitively need to improve this `NeedsUpdate` function.

Comment (sebgl) [reply operators/pkg/controller/kibana/deployment_control.go]: Yes ! I started doing it in this PR then realized it should be better off to another PR fixing issue https://github.com/elastic/cloud-on-k8s/issues/772

**Pull request merged**